### ISSUE TITLE
ForUpdate behaviour

### ISSUE THREAD

**Issue by samcook** [tumtumtum/Shaolinq#61]
How should ForUpdate work (with Postgres specifically)?

This works as expected:
`dataModel.Categories.Where(x => x.Id == categoryId).ForUpdate().ToList().First();`
```
SELECT "T0"."CategoryId", "T0"."Name"
FROM "Category" AS "T0"
WHERE (("T0"."CategoryId") = ('28ac0432-a4ca-409b-9c9c-510e7d917d93')) FOR UPDATE
```
-
These don't:
`dataModel.Categories.ForUpdate().Where(x => x.Id == categoryId).ToList().First();`
```
SELECT "T0"."CategoryId", "T0"."Name"
FROM "Category" AS "T0"
WHERE (("T0"."CategoryId") = ('28ac0432-a4ca-409b-9c9c-510e7d917d93'))
```
-
`dataModel.Categories.Where(x => x.Id == categoryId).ForUpdate().First();` or
`dataModel.Categories.ForUpdate().First(x => x.Id == categoryId);`
```
SELECT "T0"."CategoryId", "T0"."Name"
FROM "Category" AS "T0"
WHERE (("T0"."CategoryId") = ('28ac0432-a4ca-409b-9c9c-510e7d917d93')) LIMIT 1
```

**Comment by tumtumtum** [comment]
They should all work. I think the forupdate flag is being (accidentally) lost during tree rewriting with those other calls.

**Comment by tumtumtum** [comment]
This has been fixed in 50d26cb55717df6151a1abd76530828653bd47e6
